arangod/Graph/Graph.cpp

```diff
@@ -409,6 +439,26 @@ void Graph::toPersistence(VPackBuilder& builder) const {
   builder.close();  // Orphans
 }
 
+void Graph::toPersistenceWithDetails(VPackBuilder& builder,
+                                     TRI_vocbase_t& vocbase) const {
+  toPersistence(builder, true);
+
+  // calculate md5 checksum based on the graph itself
+  if (isSmart()) {
+    // additionally, check whether there are any collections created as
+    // satellite collections (Hybrid Smart Graph case).
+    auto satellites = vertexCollectionsUsedAsSatellites(vocbase);
+    builder.add(VPackValue(StaticStrings::GraphSatellites));
+
+    builder.openArray();
+    for (auto const& satCollection : satellites) {
+      builder.add(VPackValue(satCollection));
+    }
+    builder.close();
+  }
+  builder.add(StaticStrings::GraphChecksum, VPackValue(calculateMd5()));
```

Comment:
This `calculateMd5` seems to not take into account the satellites.
Ist it guaranteed to be a different value if we have two graphs that have exactly same collection names, but one graph uses one (or more) of the vertex-collections as satellite?